cluster/engine.go

```diff
@@ -513,9 +521,13 @@ func (e *Engine) updateSpecs() error {
 		// since "node" in constraint will match node.Name instead of label.
 		// Log warn message in this case.
 		if kv[0] == "node" {
-			log.Warnf("Engine (ID: %s, Addr: %s) containers a label (%s) with key of \"node\" which cannot be used in Swarm.", e.ID, e.Addr, label)
+			log.Warnf("Engine (ID: %s, Addr: %s) contains a label (%s) with key of \"node\" which cannot be used in Swarm.", e.ID, e.Addr, label)
+			continue
 		}
 
+		if value, exist := e.Labels[kv[0]]; exist {
+			log.Warnf("Engine (ID: %s, Addr: %s) already contains a label (%s) with key (%s), and Swarm will override it with docker engine label (%s)", e.ID, e.Addr, value, kv[0], kv[1])
```

Comment:
I think it better reject such labels. It's very likely a user mistake, e.g., user shouldn't set 'opertingsystem'.